### PR TITLE
fix: align btoa with wpt tests

### PR DESCRIPTION
**Pull request by caspervonb**
Part of #9001

### CODE REVIEW

**Comment by caspervonb** [comment]
Currently failing with "unsupported test environment".

**Comment by caspervonb** [comment]
Blocked on https://github.com/denoland/deno/pull/9015

**Comment by lucacasonato** [comment]
@caspervonb is this ready?

**Comment by caspervonb** [comment]
> @caspervonb is this ready?

Been waiting on the `code` property to land for DOMException, didn't realize that it landed a week ago.
Let me merge and sort out the rest of the failures, if any.

**Comment by caspervonb** [comment]
Fixed the current bugs but one of the tests fetches a json dataset which is failing (https://github.com/denoland/deno/pull/9046).

Tried filtering it out but it didn't take, not quite sure how the matching is done.

**Comment by caspervonb** [comment]
Blocked on #9252

**Comment by caspervonb** [comment]
Blocked on https://github.com/denoland/deno/issues/9266

File: op_crates/web/08_text_encoding.js [hunk 1]
```diff
@@ -124,13 +124,15 @@
...
   function btoa(s) {
+    s = String(s);
     const byteArray = [];
     for (let i = 0; i < s.length; i++) {
       const charCode = s[i].charCodeAt(0);
       if (charCode > 0xff) {
-        throw new TypeError(
+        throw new DOMException(
           "The string to be encoded contains characters " +
             "outside of the Latin1 range.",
```
Comment (lucacasonato) [inline op_crates/web/08_text_encoding.js:134]: Unrelated, but why is this a string concat?

File: op_crates/web/08_text_encoding.js [hunk 2]
```diff
@@ -124,13 +124,15 @@
   }
 
   function btoa(s) {
+    s = String(s);
```
Comment (lucacasonato) [inline op_crates/web/08_text_encoding.js:127]: At top of file:

```js
const webidl = window.__bootstrap.webidl;
```

```suggestion
    s = webidl.converters.DOMString(s, {
      prefix: "Failed to execute 'bota'",
      context: "Argument 1"
    });
```

File: op_crates/web/08_text_encoding.js (hunk 1 above)
Comment (caspervonb) [reply op_crates/web/08_text_encoding.js]: An attempt at formatting was made?
Collapsed them.

**Review by lucacasonato — APPROVED** [review]
LGTM. Thanks

**Pull request merged**